### PR TITLE
feat: Add `aws-ruby-step-functions-with-callback` example

### PR DESCRIPTION
**Pull request by Pigius**
# Big picture behind

Currently, there aren't many examples of using the serverless approach with Ruby for AWS. Ruby is a great and mature language with a great community and lots of gems.

# General message

There are not many examples of Step Functions, and especially with the Callback pattern. 

Luckily we also now have Workflow Studio for AWS Step Functions, so it's easier to design state machines.

This is an example of using `AWS Step Functions` with a callback pattern. It uses `AWS Lambda`, `DynamoDB`, `API Gateway`, `Amazon Comprehend`, flows from Step Functions.

his Workflow is quite simple. Assuming we have a comment system in our application, we would like to check for PII data before adding the comment to the database.

To do this, we use `Amazon Comprehend` which, based on trained ML models, checks if a given unstructured text contains sensitive data and returns information on which positions in the sentence it is located.

When detecting sensitive data with `Amazon Comprehend`, we use the `taskToken` from AWS step functions to wait for the detection results and continue our workflow.

Keep in mind, that taskToken is generated by Step Functions automatically.

Then, depending on the detection results, we perform a redaction process, replacing the sensitive data with an asterisk (*)

At the very end, the redacted comment is saved into the database (DynamoDB)

Diagrams could be found here:
step-functions-with-callback
stepfunctions-with-callback-graph
detailed-diagram-callback-sf

Extra points:
- Used the Ruby.27 runtime, as for the Ruby2.5 will be End of Support at the end of August.

### CODE REVIEW

File: aws-ruby-step-functions-with-callback/README.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+layers:
+  gem: arn:aws:lambda:your-region:XXXXXXXXXXX:layer:aws-ruby-step-functions-with-callback-dev-ruby-bundle:49
+Serverless StepFunctions OutPuts
+endpoints:
+  POST - https://XXXXXXXXXXXX.execute-api.your-region.amazonaws.com/dev/comments/add
+```
+
+## Usage
+
+There are two possible of usage:
```
Comment (pgrzesik) [inline aws-ruby-step-functions-with-callback/README.md:105]: Maybe rephrase this as `There are two possible ways of invoking the example` or something along those lines?

File: aws-ruby-step-functions-with-callback/README.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  "comment": "Good morning, everybody. My name is Van Bokhorst Serdar, and today I feel like sharing a whole lot of personal information with you. Let's start with my Email address SerdarvanBokhorst@dayrep.com. My address is 2657 Koontz Lane, Los Angeles, CA. My phone number is 818-828-6231. My Social security number is 548-95-6370. My Bank account number is 940517528812 and routing number 195991012. My credit card number is 5534816011668430, Expiration Date 6/1/2022, my C V V code is 121, and my pin 123456. Well, I think that's it. You know a whole lot about me. And I hope that Amazon comprehend is doing a good job at identifying PII entities so you can redact my personal information away from this document. Let's check.",
+  "author": "Daniel"
+}
+```
+As a response you will get:
+
+```json
+{
+    "executionArn": "arn:aws:states:your-region:XXXXXXXXXXXX:execution:workflow-with-callback-dev:79fa0214-8533-4506-b47f-513ca8721fe0",
+    "startDate": 1.628965530156E9
```
Comment (pgrzesik) [inline aws-ruby-step-functions-with-callback/README.md:133]: Nitpick: Let's change formatting of this value as I'm assuming it's timestamp, right?

**Review by pgrzesik — COMMENTED** [review]
Thanks a lot @Pigius :bow: I only have very minor comments, otherwise it looks great :+1:

File: aws-ruby-step-functions-with-callback/README.md (hunk 1 above)
Comment (Pigius) [reply aws-ruby-step-functions-with-callback/README.md]: Fixed!

File: aws-ruby-step-functions-with-callback/README.md (hunk 2 above)
Comment (Pigius) [reply aws-ruby-step-functions-with-callback/README.md:133]: It might be strange, but this is a response from the AWS Step Functions where you will trigger it: https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-api-gateway.html

image

Comment (pgrzesik) [reply aws-ruby-step-functions-with-callback/README.md:133]: Thanks for clarification, let's keep it this way then :+1:

**Review by pgrzesik — APPROVED** [review]
great job sir :raised_hands:
